File: Ejercicios Propuestos 1/Ejercicio 7/EJERCICIO_7.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

double random_func() {
    return static_cast<double>(rand() % 1000) / 1000.0;
}

int parte_entera(double numero_real) {
    return static_cast<int>(numero_real);
}

int main() {
    srand(time(0));

    int MAXIMO = 100;
    int numero_secreto, intento, contador;
    double x;

    x = random_func();
    numero_secreto = parte_entera(x * MAXIMO) + 1;

    contador = 0;
    intento = 0;

    cout << "Adivina el número entre 1 y " << MAXIMO << endl;

    while (intento != numero_secreto) {
        cout << "Introduce tu número:" << endl;
        cin >> intento;
        contador = contador + 1;

        if (intento < numero_secreto) {
            cout << "El número es mayor" << endl;
        } else {
            if (intento > numero_secreto) {
                cout << "El número es menor" << endl;
            }
        }
    }

    cout << "Acertaste!" << endl;
    cout << "Número de intentos: " << contador << endl;

    return 0;
}

File: Ejercicios Propuestos 1/Ejercicio 7/Ejercicio 7.jl
function random()
    return rand()
end

function parte_entera(numero_real)
    return trunc(Int, numero_real)
end

MAXIMO = 100
x = random()
numero_secret = parte_entera(x * MAXIMO) + 1

contador = 0
intento = 0

println("Adivina el número entre 1 y 100")

while intento != numero_secret
    print("Introduce tu número: ")
    global intento = parse(Int, readline())
    global contador += 1
    
    if intento < numero_secret
        println("El número es mayor")
    elseif intento > numero_secret
        println("El número es menor")
    end
end

println("Acertaste!")
println("Número de intentos: ", contador)
